src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -316,8 +324,9 @@ int halide_hexagon_remote_run(handle_t module_ptr, handle_t function,
 }
 
 int halide_hexagon_remote_poll_log(char *out, int size, int *read_size) {
+    // Read one line at a time.
     // Leave room for appending a null terminator.
-    *read_size = global_log.read(out, size - 1);
+    *read_size = global_log.read(out, size - 1, '\n');
     out[*read_size - 1] = 0;
```

Comment:
I misunderstood, I thought you were asking if read_size was not NULL, now I see you are pointing out that *read_size might be zero, which is clearly a horrible bug (I should just give up now...!). I think the - 1 simply shouldn't be there, that's overwriting the last character of the string that was read (normally a newline, which at this point doesn't matter anymore, hence no visible impact). I'm testing it now.